pkg/registry/core/pod/etcd/eviction.go

```diff
@@ -116,6 +116,10 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 }
 
 func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb policy.PodDisruptionBudget) (ok bool, err error) {
+	if pdb.Status.ObservedGeneration == nil || *pdb.Status.ObservedGeneration != pdb.Generation {
```

Comment:
Not really. If the unconditionalUpdate is false they we go into "else" branch:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/generic/registry/store.go#L408
where we face this "if":
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/generic/registry/store.go#L421